ddtrace/internal/writer/writer.py

```diff
@@ -15,6 +15,7 @@
 
 import ddtrace
 from ddtrace import config
+from ddtrace.contrib.internal.ray import in_ray_job
```

Comment:
`ddtrace/internal/writer` should not depend on modules in `ddtrace/contrib/`. All core library components should be completely decoupled from integrations otherwise we increase the likeihood of running into circular imports. Can we move `in_ray_job` to a `ddtrace/internal/distributed_computing/utils.py` module (or something similar). 

On that note, we should try to introduce distributed_computing as a product (similar to DSM, Source Code Integration, civis, etc.) and a `ddtrace.internal.distributed_computing`. This package should avoid explicit references to ray. Then in ddtrace/contrib/internal/ray we should include ray specific logic.